### ISSUE TITLE
TF not able to set `remote_repo_layout_ref` to `null`

### ISSUE THREAD

**Issue by nbaju1** [jfrog/terraform-provider-artifactory#746]
**Describe the bug**
"Remote Layout Mapping" on remote repositories has been manually set in the UI. In the UI it is not possible to set this back to blank with the following error message: "
Failed to update repository <repo-name>: Cannot invoke "org.artifactory.descriptor.repo.RepoLayout.getName()" because the return value of "org.artifactory.descriptor.config.CentralConfigDescriptor.getRepoLayout(String)" is null"

In our TF config `remote_repo_layout_ref` is not set on remote repositories, so `terraform plan` shows this:
```terraform
  # module.repositories.artifactory_remote_nuget_repository.nuget_remote will be updated in-place
  ~ resource "artifactory_remote_nuget_repository" "nuget_remote" {
        id                                    = "nuget-remote"
      - remote_repo_layout_ref                = "nuget-default" -> null
        # (36 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

`terraform apply` is not able to make this change, most likely due to the same error as in the UI. This leads to the plan showing this change on our remote repositories every time we make any changes to our Artifactory config.

Terraform: 1.4.2
Artifactory provider: 7.2.1
Artifactory version: Url for showing version does not work.

**Expected behavior**
Apply the change defined in the plan.

**Comment by alexhung** [comment]
@nbaju1 Thanks for the report. We will investigate and schedule the fix to be added.

**Comment by danielmkn** [comment]
Released in v8.2.1
Note: the provider mimics UI behavior, so once that attribute was added to the repo, it can't be removed by removing the attribute or setting it to `""`. The provider will error out on the update.
User can still create the repo without that attribute. Also, users can update all the rest attributes, without adding `remote-repo-layout-ref` if it was absent initially.
